luaui/Widgets/gui_top_bar.lua

```diff
@@ -1,19 +1,310 @@
 function widget:GetInfo()
 	return {
-		name = "Top Bar",
-		desc = "Shows Resources, wind speed, commander counter, and various options.",
-		author = "Floris",
+		name = "Top Bar with Buildpower 0.3.02",
+		desc = "Shows resources, buildpower, wind speed, commander counter, and various options.",
+		author = "Floris, Robert82 and McDoodle ",
 		date = "Feb, 2017",
-		license = "GNU GPL, v2 or later",
-		layer = -999999,
-		enabled = true,
+		license = "GNU GPL, v2 or later", 
+		layer = -99980,
+		enabled = true, --enabled by default
 		handler = true, --can use widgetHandler:x()
 	}
 end
 
+
+
+-------------------------------CONFIG------------------------------------------
+-------------------------------------------------------------------------------
+
+
+-- for bp bar only
+local config = {
+	proMode = false,
+	drawBPBar = true,
+	autoHideButtons = false,
+	debugTooltip = false,
+
+	-- Buildpower bar: behavior tweaks to consider making permanent _or_ options
+	guardingIdleBuilderCountsAsIdle = true, -- should builders guarding idle builders be considered idle themselves?
+
+	-- Show markers on the buildpower bar that indicate how much buildpower our metal and energy income could support.
+	drawBPIndicators = true,
+	drawBPWindRangeIndicators = false,
+
+	barScale = 1,
+	barWidth = 1,
+}
+
+-- used to identify when config values have changed
+local prevConfig = {}
+
+local OPTION_SPECS = {
+	{
+		configVariable = "drawBPBar",
+		name = "buildpower bar",
+		description = "Draw the buildpower bar (requires reload)",
+		type = "bool",
+	},
+	{
+		configVariable = "proMode",
+		name = "pro mode",
+		description = "Show advanced buildpower statistics and debug info (requires reload)",
+		type = "bool",
+	},
+	{
+		configVariable = "drawBPIndicators",
+		name = "eco support indicators",
+		description = "Estimate how much buildpower is supported by your metal and energy income",
+		type = "bool",
+	},
+	{
+		configVariable = "drawBPWindRangeIndicators",
+		name = "wind range indicators",
+		description = "Show how energy-supported buildpower might vary with wind speed",
+		type = "bool",
+	},
+	{
+		configVariable = "autoHideButtons",
+		name = "auto hide buttons",
+		description = "",
+		type = "bool",
+	},
+	{
+		configVariable = "debugTooltip",
+		name = "debug data",
+		description = "Show debug data in tooltip when hovering over the buildpower bar",
+		type = "bool",
+	},
+	{
+		configVariable = "barScale",
+		name = "top bar height",
+		description = "height of the top bar (requires reload)",
+		type = "slider",
+		min = 0.5,
+		max = 2,
+		step = 0.01,
+		value = 1,
+	},
+	{
+		configVariable = "barWidth",
+		name = "top bar width",
+		description = "width of the top bar (requires reload)",
+		type = "slider",
+		min = 1,
+		max = 3,
+		step = 0.01,
+		value = 1,
+	},
+}
+
+local function getOptionId(optionSpec)
+	return "top_bar_bp_" .. optionSpec.configVariable
+end
+
+local function getWidgetName()
```

Comment:
For one use, this doesnt seem worth it

Reply:
thx for pointing this one out.
Deleting it.

Reply:
fixed version is online